server/apps/publish/archive_publish_tests.py

```diff
@@ -811,40 +773,95 @@ def get_publish_items(item_id, last_version):
             self.assertEqual(published_doc[LAST_PUBLISHED_VERSION], True)
             self.assertEqual(published_digital_doc[LAST_PUBLISHED_VERSION], True)
 
-            get_resource_service('archive_correct').patch(id=doc[config.ID_FIELD], updates={'state': 'corrected'})
-            queue_items = self.app.data.find('publish_queue', None, None)
+            get_resource_service(ARCHIVE_CORRECT).patch(id=doc[config.ID_FIELD],
+                                                        updates={ITEM_STATE: CONTENT_STATE.CORRECTED})
+            queue_items = self.app.data.find(PUBLISH_QUEUE, None, None)
             self.assertEqual(10, queue_items.count())
-            published_items = self.app.data.find('published', None, None)
+            published_items = self.app.data.find(PUBLISHED, None, None)
             self.assertEqual(4, published_items.count())
             last_published_digital = get_publish_items(published_digital_doc['item_id'], True)
             self.assertEqual(1, last_published_digital.count())
             last_published = get_publish_items(published_doc['item_id'], True)
             self.assertEqual(1, last_published.count())
 
-    def test_versions_across_collections_after_publish(self):
-        with self.app.app_context():
-            self.app.data.insert('archive_versions', self.article_versions)
+    def _move_to_archived_and_assert_can_remove_from_production(self, item_id, assert_function, item_to_assert=None):
+        published_service = get_resource_service(PUBLISHED)
+        published_item = list(published_service.get_from_mongo(req=None, lookup={'item_id': item_id}))
+        self.assertEquals(len(published_item), 1)
 
-            # Publishing an Article
+        # Moving to archived explicitly
+        published_item = published_item[0]
+        published_service.patch(id=published_item[config.ID_FIELD], updates={'allow_post_publish_actions': False})
+
+        assert_function(published_service.can_remove_from_production(
+            item_to_assert if item_to_assert else published_item))
+
+        return published_item
+
+    def test_can_remove_from_production_first_rule(self):
+        """
+        Test for can_remove_production() for the below:
+            1. When item is published only once. In this case can_remove_production() should return true
+            2. When item is published more than once. In this case can_remove_production() should return false
```

Comment:
I would suggest splitting this into two separate tests, each of them testing only a single case.